### DISCUSSION TITLE
Why?

### DISCUSSION THREAD

**Discussion by Imborked** [builtbybel/Bloatynosy#165] [General]
Why would you turn your app into the same kind of trash I use your app to get rid of? 

I just need a bloat remover, I don't need my bloat remover to also be bloated with the same trash I'm trying to remove from my windows installs.

**Comment by ilianoKokoro** [comment]
The new UI really hinders the use of the app. I can't even figure out how to use the features of the original. Maybe I'm missing something, but this is now a different app with a different purpose that I no longer have a use for. Even just adding a menu to switch from the new app to the manual de-bloating of the old app would fix the problem.

**Comment by Broeckelmann** [comment]
Intuitive Usage is basically gone .-.

It was so easy beforehand, but now...

**Comment by ScottRFrost** [comment]
As an end-user I want to enable the full context menu.

Old bloaty nosy: Open System, double click Enable full context, click apply.

New bloaty nosy: Scroll down and look for option, can't find it, type in "context" to the dumb AI box, can't find it, type in "menu" to the dumb AI box, can't find it, give up, go back to old bloaty nosy.

GIANT step backwards in usability.

**Comment by Belim** [comment]
Hey guys, I know you can't please everyone, but the complaints were often present even in the last 1.x Bloatynosy Classic version in the issues here. Some cope with it, others don't. I will migrate all settings from Bloatynosy Classic to Winpilot. The [current 3.05 version already brings a lot with it](https://github.com/builtbybel/Winpilot/releases/tag/3.0.5).

![Winpilot_VvUYtiXwk6](https://github.com/builtbybel/Winpilot/assets/57478606/ecd626f0-9b08-4c50-83fa-3ecff3b465e4)
Here, it really doesn't get any easier. You can either pull up the current settings or make adjustments yourself and then enable them with just one click. Just like in Bloatynosy, with a simple click.

**Comment by A9G-Data-Droid** [comment]
When I read the new README, my gut reaction was "y'all jumped the shark". I know, first impressions like this can be harsh but I have to agree with the general sentiment here. 

Let me be clear: WinPilot is cool. I like it for what it is but its' a separate product from BloatyNosy. If I want a debloater, I would use BloatyNosy. If I wanted an AI assistant who can change windows settings I would use WinPilot.

Even the fact that WinPilot can call upon Bloaty Nosy is cool. They just need to be kept separate.

Modular software is good. [To quote the old grey beards of Unix](https://en.wikipedia.org/wiki/Unix_philosophy):

> Make each program do one thing well. To do a new job, build afresh rather than complicate old programs by adding new "features".
> Expect the output of every program to become the input to another, as yet unknown, program. Don't clutter output with extraneous information. Avoid stringently columnar or binary input formats. Don't insist on interactive input.

**Comment by Belim** [comment]
> When I read the new README, my gut reaction was "y'all jumped the shark". I know, first impressions like this can be harsh but I have to agree with the general sentiment here.
> 
> Let me be clear: WinPilot is cool. I like it for what it is but its' a separate product from BloatyNosy. If I want a debloater, I would use BloatyNosy. If I wanted an AI assistant who can change windows settings I would use WinPilot.
> 
> Even the fact that WinPilot can call upon Bloaty Nosy is cool. They just need to be kept separate.
> 
> Modular software is good. [To quote the old grey beards of Unix](https://en.wikipedia.org/wiki/Unix_philosophy):
> 
> > Make each program do one thing well. To do a new job, build afresh rather than complicate old programs by adding new "features".
> > Expect the output of every program to become the input to another, as yet unknown, program. Don't clutter output with extraneous information. Avoid stringently columnar or binary input formats. Don't insist on interactive input.

Bloatynosy had become ingrained, and experienced users associated it with hardcore tuning and debloating. Over the past few months, I've addressed some of the issues, and today, I've closed them all. Many users either broke something or couldn't handle the functions. I had mixed feelings about it. You have no idea how many emails I received daily. I eventually shut down my email address. With Winpilot, I'm moving in a different direction, focusing on usability inspired by Microsoft. That's the way things are going. Microsoft is forcing the Copilot upon us; soon, even Auntie and Grandpa will know what it is and what it does because it will be present in every conceivable window and context menu. Winpilot shouldn't confuse users like Bloatynosy did because of its name; it should be clear what it does. Modeled after the Copilot, it guides Windows for the better and still offers smart debloating for the average user. The plugin CoTweaker encompasses almost all functions of Bloatynosy. What is missing here? Is it just the name? The plugin could also be called Bloatynosy... The first 5 plugins cover everything that Bloatynosy could do. The Assisted chatbox allows frequent settings adjustments to be made quickly without having to set checkboxes or move entries from one list to another, as in Bloatynosy.

**Comment by A9G-Data-Droid** [comment]
I apologize, I don't know the history. I'm just saying they are very different things. To me it's similar to "You want a car? Okay, here is the truck that delivers cars. Drive the truck to the your destination, park, and then unload the car you want from it". When before it was just a single car, the one people came here for.

If I want to debloat 10 computers, do I have to talk to an AI 10 times to do so? I would much prefer a tool that does one clear thing and is scriptable. Save your settings, apply your settings. It should be shipped as a single, stand-alone binary that doesn't need to be installed. Then it can be called by a login script with a settings file and apply the saved settings automatically.

This form factor could also be called by WinPilot. Some thing that both humans and AIs can enjoy.

**Comment by Wolfhound905** [comment]
I miss the UX of thisiswin11

**Comment by LyndonGingerich** [comment]
Can Winpilot really do everything that BloatyNosy could? I remember seeing a lot more tweak options in BloatyNosy. And I'm not seeing a way to add plugins to Winpilot.

**Comment by Belim** [comment]
> Can Winpilot really do everything that BloatyNosy could? I remember seeing a lot more tweak options in BloatyNosy. And I'm not seeing a way to add plugins to Winpilot.

The staticAIData file in the apps directory supports dozens of skills, including executing PowerShell scripts, reg.exe commands, a plethora of command-line parameters, or integration with third-party tools. Indeed, the CoTweaker is still missing a few options from Bloatynosy, but I'll migrate those too. Do you have a specific one in mind?

**Comment by Belim** [comment]
Feel free to continue the [discussion](https://github.com/orgs/builtbybel/discussions) there

**Comment by LyndonGingerich** [comment]
> > Can Winpilot really do everything that BloatyNosy could? I remember seeing a lot more tweak options in BloatyNosy. And I'm not seeing a way to add plugins to Winpilot.
> 
> The staticAIData file in the apps directory supports dozens of skills, including executing PowerShell scripts, reg.exe commands, a plethora of command-line parameters, or integration with third-party tools. Indeed, the CoTweaker is still missing a few options from Bloatynosy, but I'll migrate those too. Do you have a specific one in mind?

No.

It looks to me as though this UI change has mostly added AI and Clippy. As far as I know, programmers (or at least me and The Register) generally object to AI's introduction everywhere. And isn't Clippy such a meme specifically because he's universally despised? I'm not understanding how the UI change helps users--nor why it reduced your emails.

Have you thought of looking for co-maintainers for the old product and requiring issues to be opened on GitHub rather than sent to your individual email address?

**Comment by ScottRFrost** [comment]
In searching for an alternative, the best I've found so far is https://winaerotweaker.com, but the UI isn't great and it's not open source.  It's certainly much better than WinPilot UI though.

**Reply by A9G-Data-Droid** [reply]
It would be better to fork the old BloatyNosy then to switch to this.

**Reply by ScottRFrost** [reply]
I'd be OK with that, honestly, if someone wanted to take it up

**Reply by LyndonGingerich** [reply]
If all we need is someone to wait for someone more enthusiastic to hand the fork off to, I can do that: https://github.com/LyndonGingerich/BloatyNosy

If @Belim wants it back at any point, he may have it.

**Reply by ScottRFrost** [reply]
Long live (pre awful UI choices) BloatyNosy!

**Reply by Belim** [reply]
Did anyone actually bother to properly test and actively use the latest version (3.4)? Feel free to voice any criticisms about the UI and features. Do you want Winpilot with the classic Bloatynosy UI? That's the only thing I seem to keep hearing about, the UI? Nearly 90% of the features from Bloatynosy are available, along with dozens of others you can utilize, even beyond the "Assisted/AI"chatbox you guys keep mentioning. In fact, in the current version 3.4, the chatbox has even taken a backseat. I've bid farewell to the name Bloatynosy, and that's final. I'll continue actively developing Winpilot and I'm open to implementing suggestions.

**Comment by LyndonGingerich** [comment]
@Belim I'll test 4.3.0.

My issues with the UI:
- AI. Winpilot does not need AI; its menu system is clear. If you know what you want to do and how to do it, why would you take the risk of the AI misunderstanding you?
- Clippy. His speech balloons cover what I'm trying to do. I understand his memeability, but he doesn't make the product better; rather, he interferes with its use.
- Didn't BloatyNosy provide a platform for people to write and distribute their own plugins? I'm pretty sure I was using one from ChrisTitusTech. I haven't seen that in Winpilot. (EDIT: Apparently I meant WinModder. WinModder was cool.)
- Winpilot feels childproofed somehow compared to BloatyNosy.
- BloatyNosy didn't have profanity (though I might misremember).
- Out of the screens in Winpilot, CoTweaker and Decrapify make sense.
  - WingetUI has been done repeatedly for users who don't know what a terminal is. Typing is faster than scrolling, so even if I wanted a mouse interface for winget, I would pick one with a search bar. WingetUI provides only a list of suggested programs to install. But people who don't know what they want to install aren't going to find Winpilot in the first place.
  - Copilotless seems like it should be under Decrapify.
  - ClippySupreme is, well, Clippy.

**Reply by LyndonGingerich** [reply]
I use Winpilot specifically because I want to modify my system in risky ways. I want to examine what I'm doing very precisely. AI hides detail and adds a level of unpredictability. A slight miscommunication could lead to a system modification requiring reinstall of Windows or something. I don't want AI for anything for which I use Winpilot.

**Reply by LyndonGingerich** [reply]
I've reported a few of my more specific UI issues.

**Reply by LyndonGingerich** [reply]
With BloatyNosy, the longer I played with it, the more cool features I found to try out; I liked it even though it was buggy. With Winpilot, the longer I mess with it, the more frustrated I grow. Most of it is of no use to me.

And are you certain that the CoTweaker list is as long as the list of tweaks in BloatyNosy? BloatyNosy's seemed longer.

**Reply by LyndonGingerich** [reply]
@Belim, I sympathize with your feeling overwhelmed by the workload of BloatyNosy. Some of those old issues look rather newbie and low-effort. But maybe the community could have helped more? I didn't even know you were feeling overwhelmed. (I personally probably wouldn't have helped, but if I didn't know, then possibly others didn't, either.)

**Comment by Belim** [comment]
Thanks for the detailed pointers and the flurry of bug reports @LyndonGingerich , I can work with that.. The smart approach in Winpilot is still strongly represented, but nothing is left to chance. There are no fully autonomous actions, everything is "assisted". The upcoming version will also only intelligently assist in the search for bloatware, but won't utilize true AI. Winpilot should be the counterpart to the Copilot, but exclusively for Windows. I needed a mascot that reports on the success of actions and even lends a hand to the user. Clippy seemed fitting for that role (at first, I was considering Cortana and even Microsoft BOB). Clippy won't disappear anymore, he's too deeply rooted in the app. If anything, Winpilot would disappear before Clippy does. There are numerous plugins available, but not visible to the naked eye and must be accessed through the chatbox. I can revive WinModder for the future and display all plugins in an organized area with the option to add new plugins and extensions. Especially your point about discovering new things with Bloatynosy but not with Winpilot, I can't understand, since in Winpilot everything is modular and limitless. Just through the chatbox, you could access 50 more actions than in Bloatynosy. As I said, the decision has already been made. Winpilot will be further developed, it's just the beginning, and sooner or later, I hope it will also convince you. I wanted to create an app that can truly be used by everyone without any strange aftertaste, and unfortunately, that was always the case with Bloatynosy. By the way, I even got laughs from Microsoft colleagues for the name Bloatynosy ;)

**Reply by LyndonGingerich** [reply]
> The smart approach in Winpilot is still strongly represented, but nothing is left to chance. There are no fully autonomous actions, everything is "assisted". The upcoming version will also only intelligently assist in the search for bloatware, but won't utilize true AI.

Wait, let me see whether I understand correctly: The "chat" in Winpilot is just a feature search? If so, I think I see where this is going wrong:
- Many are fighting AI because of its unreliability, so Winpilot is receiving undeserved flak because it calls itself an AI app.
- A Windows mod and cleanup tool should not need AI for anything. Adding AI sounds like exactly the random AI injection people are distrusting from Microsoft (such as in Paint and Notepad). So again, Winpilot is receiving undeserved flak because it says that it is using AI when it is not.
- If features are accessible only from chat, then no one will know what they are without hunting down `staticAIData.json`, or stumbling upon it as I did recently. So Winpilot looks underpowered since people will see only the listed plugins.

Actually, what does the Clippy chatbox even do? On 3.4.0, it tells me that no results were found and redirects me to Copilot even if I just type the letter "a". Is it a command search box that requires exact matching of entire phrases? If so, what you have here is a CLI tool with a needless GUI, no documentation, and unnecessarily verbose English commands.

Come to think of it, a Winpilot CLI app with some sort of man-page-equivalent "help" command would be pretty useful.

> By the way, I even got laughs from Microsoft colleagues for the name Bloatynosy ;)

I liked it. Though I liked it better with the capitalized N to match C#'s and F#'s CamelCase.

**Reply by LyndonGingerich** [reply]
Okay, it looks as though Clippy's chat is a feature search that was just broken in 3.4.0. But I still think that all features need an index, not just a search, for discoverability.

**Reply by LyndonGingerich** [reply]
I just searched for "titus" to find Chris Titus's tool. The item that appeared said "Show Chris Titus's tool in action". I wanted to know what it did, so I clicked it, thinking that "Show" meant that it would show me what the tool does. Instead, it just ran the script without telling me anything about what it did.

I think that Winpilot definitely needs more discoverability and documentation.

**Reply by Belim** [reply]
thats why we ended up talking past each other 😼

**Comment by Belim** [comment]
It was a bit more complex.. it dates back to the BloatynosyAI era. The Decrapify Plugin was able to identify bloatware based on certain patterns. Initially, this was processed through GitHub, then locally, and now I'm back to using a local static DB. That's also why the Winpilot code hasn't been released yet (coming soon), and why I'm still tidying up. I myself struggled with the architecture in the end, at one point attempted integration with Mistral AI buut eventually gave up. Yes, Winpilot now operates entirely locally and without AI but still exhibits some smart indicators..

**Comment by ScottRFrost** [comment]
**3.5.1 Tested**

![image](https://github.com/builtbybel/Winpilot/assets/3866418/2ebc5842-b452-4888-b70e-b0a03f42c75b)
What's the point of this startup screen?  It's not clear what I'm supposed to do.

I give up on the no content start page and click Next, which gives me:
![image](https://github.com/builtbybel/Winpilot/assets/3866418/6050db96-a3b0-4ba4-a915-ca2f7e012418)

At first I'm confused by why I have to click Detect current settings instead of it doing that automatically, so I click it and it gives
![image](https://github.com/builtbybel/Winpilot/assets/3866418/61c04d69-6878-4863-be3f-fd6f1eeff57a)
WOW, I don't want ANY OF THAT.  I just wanted the checkboxes to display the current values!

So I click out of the yellow bloat box and go to toggle something to test.  I chose Uninstall Fax printer which gives me:
![image](https://github.com/builtbybel/Winpilot/assets/3866418/80747d22-b77c-43ca-b51f-fdf648b61686)
MORE stuff I don't want.  Come on, you have to be kidding me!

So just for fun I want to see if it committed my change or if I have to hit next (because there's no indication on the UI), so I click Detect current settings again.... aaaannnddd... Uninstall Fax Printer is back to off, so I guess it didn't do anything.  I give up and go back to real BloatyNosy.

**Comment by stamminator** [comment]
What is the recommendation for those of us who want a Windows 11 tuning and debloating tool with zero AI whatsoever?

**Reply by ScottRFrost** [reply]
I've found https://github.com/LeDragoX/Win-Debloat-Tools to have all the features I was looking for in BloatyNosy, but super simple and lightweight.  It's also just a big powershell script, so it's easy to look through and understand what each option is doing.  No AI crap and 50 extra pages of UI to do simple tasks like some other debloat tools.
